mdns.go

```diff
@@ -1,18 +1,41 @@
 package main
 
 import (
+	"encoding/json"
+	"fmt"
+	"io"
 	"log"
 	"net"
+	"net/http"
+	"os"
+	"time"
 
 	"github.com/grandcat/zeroconf"
 	"github.com/rs/xid"
 )
 
+type Response struct {
+	Result  string         `json:"result"`
+	Message string         `json:"message,omitempty"`
+	Data    map[string]any `json:"data,omitempty"`
+}
+
 func publishHomeAssistant() {
 	var err error
 
+	var outboundIP net.IP
+
+	for {
+		outboundIP, err = getOutboundIP()
+		if outboundIP != nil {
+			break
+		}
+		log.Printf("Failed to get outbound IP, retrying in 5s: %s", err)
+		time.Sleep(5 * time.Second)
+	}
+
 	unique := xid.New()
-	hostURL := "http://" + getOutboundIP().String() + ":8123"
+	hostURL := "http://" + outboundIP.String() + ":8123"
```

Comment:
Same here, `net/url` would make this a bit prettier it feels.

Reply:
Well, looked into that and I don't see it much nicer, you still need to concat the port, so `hostURL := "http://" + outboundIP.String() + ":8123"` turns into:
```
hostURL := &url.URL{
    Scheme: "http",
    Host:   outboundIP.String() + ":8123",
}
...
hostURL.String()
```

If there's no other concern than making it pretty, I'd stick with what we did before.